exarl/base/agent_base.py

```diff
@@ -52,37 +54,56 @@ def get_weights(self):
         pass
 
     @abstractmethod
-    def set_weights(self):
+    def set_weights(self, weights):
         """set target model weights
         """
         pass
 
     @abstractmethod
-    def train(self):
+    def train(self, batch):
         """train the agent
         """
         pass
 
     @abstractmethod
-    def action(self):
+    def target_train(self):
```

Comment:
I am not sure if we should mandate target_train as some algorithms use a single active policy. SMEs should comment on this.